tests/test_gridsearch.py

```diff
@@ -702,6 +705,18 @@ def test_local_mode(self):
         self.assertTrue(hasattr(grid_search, "cv_results_"))
         self.assertTrue(wrapped_init.call_args[1]["local_mode"])
 
+    def test_tune_search_spaces(self):
+        # Test mixed search spaces
+        clf = MockClassifier()
+        grid_search = TuneGridSearchCV(
+            clf, {
+                "foo_param": tune.grid_search([1, 2, 3]),
+                "bar_param": [1, 2]
+            },
+            refit=False,
+            cv=3)
+        grid_search.fit(X, y)
+
```

Comment:
Should this test if all combinations were seen?

Reply:
Good call, added